waku/v2/node/wakunode2.nim

```diff
@@ -652,195 +643,283 @@ proc connectToNodes*(n: WakuNode, nodes: seq[PeerInfo]) {.async.} =
   # later.
   await sleepAsync(5.seconds)
 
+proc start*(node: WakuNode) {.async.} =
+  ## Starts a created Waku Node and
+  ## all its mounted protocols.
+  ##
+  ## Status: Implemented.
+  
+  node.libp2pTransportLoops = await node.switch.start()
+  
+  # TODO Get this from WakuNode obj
+  let peerInfo = node.peerInfo
+  info "PeerInfo", peerId = peerInfo.peerId, addrs = peerInfo.addrs
+  let listenStr = $peerInfo.addrs[^1] & "/p2p/" & $peerInfo.peerId
+  ## XXX: this should be /ip4..., / stripped?
+  info "Listening on", full = listenStr
+
+  if not node.wakuRelay.isNil:
+    await node.startRelay()
+  
+  info "Node started successfully"
+  node.started = true
+
+proc stop*(node: WakuNode) {.async.} =
+  if not node.wakuRelay.isNil:
+    await node.wakuRelay.stop()
+
+  await node.switch.stop()
+
+  node.started = false
+
 {.pop.} # @TODO confutils.nim(775, 17) Error: can raise an unlisted exception: ref IOError
 when isMainModule:
+  ## Node setup happens in 6 phases:
+  ## 1. Set up storage
+  ## 2. Initialize node
+  ## 3. Mount and initialize configured protocols
+  ## 4. Start node and mounted protocols
+  ## 5. Start monitoring tools and external interfaces
+  ## 6. Setup graceful shutdown hooks
+
   import
+    confutils,
     system/ansi_c,
-    confutils, json_rpc/rpcserver, metrics,
-    ./config, 
-    ./jsonrpc/[admin_api,
-               debug_api,
-               filter_api,
-               private_api,
-               relay_api,
-               store_api],
+    ../../common/utils/nat,
+    ./config,
+    ./waku_setup,
     ./storage/message/waku_message_store,
-    ./storage/peer/waku_peer_storage,
-    ../../common/utils/nat
+    ./storage/peer/waku_peer_storage
+  
+  logScope:
+    topics = "wakunode.setup"
+  
+  ###################
+  # Setup functions #
+  ###################
+
+  # 1/6 Setup storage
+  proc setupStorage(conf: WakuNodeConf):
```

Comment:
Wondering whether it can be moved to waku_setup?

Reply:
I experimented with moving some of the phases (and not just the utilities) to `waku_setup`, but found that it caused some issues, including confusing circular dependencies. `setupStorage` _is_ a phase that could be moved, but that would impede readability somewhat (as it will be the only setup phase not in `wakunode2.nim`). WDYT? Going to merge as-is for now, but it should be easy to move things around in future.